src/main/java/com/amazonaws/services/dynamodbv2/datamodeling/encryption/DynamoDBEncryptor.java

```diff
@@ -81,7 +81,8 @@
     private final String signingAlgorithmHeader;
     
     public static final String DEFAULT_SIGNING_ALGORITHM_HEADER = DEFAULT_DESCRIPTION_BASE + "signingAlg";
-    
+    private Function<EncryptionContext, EncryptionContext> encryptionContextOverrideOperator;
```

Comment:
This should be a `UnaryOperator` here and throughout.

Reply:
I'd have liked `UnaryOperator` as well, but the chaining methods inherited from `Function<T,T>` returns a `Function<T,T>` instead of an UnaryOperator. Ex. calling `.andThen` on an `UnaryOperator<T>` returns a `Function<T, T>`.

I'm tempted to drop `UnaryOperator` completely and instead use `Function<T,T>` throughout, but also think that returning a more specific object from a helper class is acceptable. What do you think?